pkg/util/restmapper/restmapper.go

```diff
@@ -11,6 +17,98 @@ func GetGroupVersionResource(restMapper meta.RESTMapper, gvk schema.GroupVersion
 	if err != nil {
 		return schema.GroupVersionResource{}, err
 	}
-
 	return restMapping.Resource, nil
 }
+
+// cachedRESTMapper is a RESTMapper that will provides gvkToGVR cache for RESTMapping Method.
+type cachedRESTMapper struct {
+	restMapper meta.RESTMapper
+	gvkToGVR   sync.Map
+}
+
+func (g *cachedRESTMapper) KindFor(resource schema.GroupVersionResource) (schema.GroupVersionKind, error) {
+	//TODO implement me
+	return g.restMapper.KindFor(resource)
+}
+
+func (g *cachedRESTMapper) KindsFor(resource schema.GroupVersionResource) ([]schema.GroupVersionKind, error) {
+	//TODO implement me
+	return g.restMapper.KindsFor(resource)
+}
+
+func (g *cachedRESTMapper) ResourceFor(input schema.GroupVersionResource) (schema.GroupVersionResource, error) {
+	//TODO implement me
+	return g.restMapper.ResourceFor(input)
+}
+
+func (g *cachedRESTMapper) ResourcesFor(input schema.GroupVersionResource) ([]schema.GroupVersionResource, error) {
+	//TODO implement me
+	return g.restMapper.ResourcesFor(input)
+}
+
+func (g *cachedRESTMapper) RESTMappings(gk schema.GroupKind, versions ...string) ([]*meta.RESTMapping, error) {
+	//TODO implement me
+	return g.restMapper.RESTMappings(gk, versions...)
+}
+
+func (g *cachedRESTMapper) ResourceSingularizer(resource string) (singular string, err error) {
+	//TODO implement me
+	return g.restMapper.ResourceSingularizer(resource)
+}
+
+func (g *cachedRESTMapper) RESTMapping(gk schema.GroupKind, versions ...string) (*meta.RESTMapping, error) {
+	if len(versions) == 1 {
+		currGVK := gk.WithVersion(versions[0])
+		value, ok := g.gvkToGVR.Load(currGVK)
+		if !ok {
+			restMapping, err := g.restMapper.RESTMapping(gk, versions...)
+			if err != nil {
+				return restMapping, err
+			}
+			g.gvkToGVR.Store(currGVK, restMapping)
+			value = restMapping
+		}
+		return value.(*meta.RESTMapping), nil
+	}
+
+	return g.restMapper.RESTMapping(gk, versions...)
+}
```

Comment:
```golang
func (g *cachedRESTMapper) RESTMapping(gk schema.GroupKind, versions ...string) (*meta.RESTMapping, error) {
	// in case of multi-versions, cachedRESTMapper don't know which is the preferred version,
	// so just bypass the cache and consult the underlying mapper.
	if len(versions) > 1 {
		return g.restMapper.RESTMapping(gk, versions...)
	}

	if len(versions) == 0 {
		return nil, fmt.Errorf("expected at least one version")
	}

	gvk := gk.WithVersion(versions[0])
	value, ok := g.gvkToGVR.Load(gvk)
	if ok { // hit cache, just return
		return value.(*meta.RESTMapping), nil
	}

	// consult underlying mapper and then update cache
	restMapping, err := g.restMapper.RESTMapping(gk, versions...)
	if err != nil {
		return restMapping, err
	}
	g.gvkToGVR.Store(gvk, restMapping)
	return restMapping, nil
}
```